db/migrate/20211029070915_add_treatment_group_membership_constraint.rb

```diff
@@ -0,0 +1,19 @@
+class AddTreatmentGroupMembershipConstraint < ActiveRecord::Migration[5.2]
+  def up
+    patients_in_multiple_experiments =
+      Experimentation::TreatmentGroupMembership
+        .group(:patient_id, :experiment_id)
+        .having("count(*) > 1")
+        .select(:patient_id)
+
+    Experimentation::TreatmentGroupMembership
+      .where(patient_id: patients_in_multiple_experiments)
+      .destroy_all
+
+    add_index :treatment_group_memberships, [:patient_id, :experiment_id], unique: true, name: "index_tgm_patient_id_and_experiment_id"
```

Comment:
Have you thought about which column should be the first one in this index?

Reply:
Indexes on both these columns already exist, so this index shouldn't be used when just one of them is present.